typescript/src/schema/api.ts

```diff
@@ -83,3 +91,48 @@ export class API {
     );
   }
 }
+
+function getResourceMap(
+  fd: plugin.google.protobuf.IFileDescriptorProto[]
+): ResourceMap {
+  const resourceMap: ResourceMap = {};
+  for (const fileDescriptor of fd) {
+    const messages = fileDescriptor
+      .messageType!.filter(message => message.name)
+      .reduce(
+        (map, message) => {
+          map['.' + fileDescriptor.package! + '.' + message.name!] = message;
+          return map;
+        },
+        {} as MessagesMap
+      );
+    for (const property of Object.keys(messages)) {
+      const m = messages[property];
+      if (m && m.options) {
+        const option = m.options;
+        if (option && option['.google.api.resource']) {
+          const opt = option['.google.api.resource'];
+          const oneResource = option[
+            '.google.api.resource'
+          ] as ResourceDescriptor;
+          if (opt.type) {
+            const arr = opt.type.match(/\/([^.]+)$/);
+            if (arr) {
+              oneResource.name = arr[arr.length - 1];
```

Comment:
Since you only have one match group in your regex, it should always be `arr[1]`.

Reply:
yes, that makes sense, there is only one element in the array `arr[0]`.